### DISCUSSION TITLE
How to access existing applications from new scenario

### DISCUSSION THREAD

**Discussion by pv2-0** [eclipse/mosaic#49] [Q&A] [answered]
Hi,

I am trying to add Roadside Units to my simulation and wanted to use the applications from the examples as a starting point (eventually I want to add a V2X communication application as well). When I try to add a roadside unit similar to the one in the Tiergarten example with coordinates that suit my osm network. I added the RSU into my mapping file with the org.eclipse.mosaic.app.tutorial.RoadSideUnitApp application and added the RSU in cell/regions.json.
However, when I try to run it, I get the error:

> Stopping simulation due to a critical error:
>           -InternalFederateException: Error code 0x01000013: There was an error during and advance time.
>           -Root Cause: ClassNotFoundException: org.eclipse.mosaic.app.tutorial.RoadSideUnitApp
> Please see the log files for details.
>           -Log-Directory: logs/log-20210302-013538-test_dt
> Eclipse MOSAIC will now shutdown.

I checked the log file but it is empty. The Tiergarten scenario doesn't have an app for roadside units in its application folder.

Where can I find this application and how can I access it from my own scenario?

Thanks!

**Comment by kschrab** [comment, marked as answer]
This specific application (`org.eclipse.mosaic.app.tutorial.RoadSideUnitApp`) is bundled in the `traffic-light-communication-20.0.jar` file.

**Comment by yoge1983** [comment]
his specific application (org.eclipse.mosaic.app.tutorial.RoadSideUnitApp) is bundled in the traffic-light-communication-20.0.jar file.
